cachestore/memory/cache_store.go

```diff
@@ -0,0 +1,86 @@
+package memory
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/coocood/freecache"
+	"github.com/merlinapp/datarepo-go"
+	"github.com/merlinapp/datarepo-go/drreflect"
+	"log"
+	"time"
+)
+
+type memoryBasedCacheStore struct {
+	cache *freecache.Cache
+}
+
+// Creates a new CacheStore backed by a freecache Cache (github.com/coocood/freecache)
+//
+// Implementation Notes: Currently this implementation serializes the data to JSON
+// for storage in the memory cache
+func NewFreeCacheInMemoryStore(cacheSize int) datarepo.CacheStore {
+	cache := freecache.NewCache(cacheSize)
+
+	store := memoryBasedCacheStore{
+		cache: cache,
+	}
+	return &store
+}
+
+func (c *memoryBasedCacheStore) Set(ctx context.Context, key string, value interface{}, expiration time.Duration) {
+	bytesToCache, err := cacheMarshal(value)
+	if err != nil {
+		log.Println("Error setting cache value for key: ", key, "-", err)
+		return
+	}
+
+	if err = c.cache.Set([]byte(key), bytesToCache, int(expiration.Seconds())); err != nil {
+		log.Println("Error setting cache value for key: ", key, "-", err)
+	}
+}
+
+func (c *memoryBasedCacheStore) Get(ctx context.Context, key string, out interface{}) (bool, error) {
+	cachedBytes, err := c.cache.Get([]byte(key))
+	if err != nil {
+		if err == freecache.ErrNotFound {
+			return false, nil
+		}
+		return false, err
+	}
+
+	if err = cacheUnmarshal(cachedBytes, out); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func (c *memoryBasedCacheStore) Delete(ctx context.Context, key string) error {
+	c.cache.Del([]byte(key))
+	return nil
+}
+
+func (c *memoryBasedCacheStore) GetMulti(ctx context.Context, keys []string, out interface{}) ([]bool, error) {
+	found := make([]bool, len(keys))
+
+	// the out interface is expected to be of type: *[]*A assuming this cache stores elements of type A
```

Comment:
If is expected to always be a pointer of a slice on any type, it might be a good idea to change the firm of the interface to *[]interface{} in the future? to avoid confusion

Reply:
Hey,
Yeah, that's a good point. I'll add it to the issue list to keep it in the backlog.
Thanks!

Reply:
https://github.com/merlinapp/datarepo-go/issues/3